canvas_modules/common-canvas/src/notification-panel/notification-panel.jsx

```diff
@@ -106,31 +120,87 @@ class NotificationPanel extends React.Component {
 		}
 	}
 
+	deleteNotification(messageKey) {
+		this.props.canvasController.deleteNotificationMessages(messageKey);
+	}
+
+	clearNotificationMessages() {
+		this.props.canvasController.clearNotificationMessages();
+	}
+
+	closeNotificationPanel() {
+		this.props.canvasController.toolbarActionHandler("closeNotificationPanel");
+	}
+
 	render() {
 		const notificationPanelClassName = this.props.isNotificationOpen ? "" : "panel-hidden";
-		const notificationPanel = this.props.messages.length > 0
-			? (<div className="notification-panel">
-				<div className="notification-panel-header">{this.props.notificationHeader}</div>
-				<div className="notification-panel-messages">
-					{this.getNotifications()}
+		const notificationHeader = this.props.notificationConfig && this.props.notificationConfig.notificationHeader
+			? this.props.notificationConfig.notificationHeader
+			: DEFAULT_NOTIFICATION_HEADER;
+		const notificationSubtitle = this.props.notificationConfig && this.props.notificationConfig.notificationSubtitle
+			? (<div className="notification-panel-subtitle">
+				{this.props.notificationConfig.notificationSubtitle}
+			</div>)
+			: null;
+		const notificationPanelMessages = this.props.messages.length > 0
+			? this.getNotifications()
+			: (<div className="notification-panel-empty-message-container">
+				<div className="notification-panel-empty-message">
+					{this.props.notificationConfig && this.props.notificationConfig.emptyMessage ? this.props.notificationConfig.emptyMessage : null}
 				</div>
-				<svg className="notification-popup-arrow" x="0px" y="0px" viewBox="0 0 16 9">
-					<polyline points="0,9 8,0 16,9" />
-				</svg>
+			</div>);
+		const clearAll = this.props.notificationConfig && this.props.notificationConfig.clearAllMessage
+			? (<div className="notification-panel-clear-all-container">
+				<Button
+					className="notification-panel-clear-all"
+					onClick={this.clearNotificationMessages.bind(this)}
+					kind="ghost"
+					size="small"
+					disabled={this.props.messages.length === 0}
+				>
+					{this.props.notificationConfig.clearAllMessage}
+				</Button>
 			</div>)
-			: <div />;
+			: null;
 
 		return (<div className={"notification-panel-container " + notificationPanelClassName} >
-			{notificationPanel}
+			<div className="notification-panel">
+				<div className="notification-panel-header">
+					{notificationHeader}
+					{notificationSubtitle}
+					<Close16 className="notification-panel-close-icon" onClick={this.closeNotificationPanel} />
+				</div>
+				<div className="notification-panel-messages-container">
+					<div className="notification-panel-messages">
+						{notificationPanelMessages}
+					</div>
+					{clearAll}
+				</div>
+				<svg className="notification-popup-arrow" x="0px" y="0px" viewBox="0 0 16 9">
```

Comment:
The arrow should be the same color as the title background
<img width="337" alt="image" src="https://user-images.githubusercontent.com/1628100/83695367-bb179980-a5ae-11ea-9a50-9c5e751061ab.png">

Reply:
fixed